src/imputors/drop.jl

```diff
@@ -36,38 +46,134 @@ NOTES (or premature optimizations):
     3. reshaping the data back to the desired shape.
 
 # Arguments
-* `imp::Drop`: this `Imputor` method
-* `ctx::Context`: contextual information for missing data
+* `imp::DropObs`: this `Imputor` method
 * `data::AbstractMatrix`: the data to impute
 
 # Returns
 * `AbstractMatrix`: a new matrix with missing rows removed
 """
-function impute!(imp::Drop, ctx::Context, data::AbstractMatrix)
-    ctx.num = size(data, 1)
-    mask = map(i -> !ismissing(ctx, data[i, :]), 1:size(data, 1))
-    return data[mask, :]
+function impute!(imp::DropObs, data::AbstractMatrix)
+    imp.context() do c
+        mask = map(i -> !ismissing(c, data[i, :]), 1:size(data, 1))
+        return data[mask, :]
+    end
+end
+
+"""
+    impute!(imp::DropObs, table)
+
+Finds the missing rows in the table and deletes them.
+
+# Arguments
+* `imp::DropObs`: this `Imputor` method
+* `table`: a type that implements the Tables API.
+
+# Returns
+* our data with the missing rows removed.
+"""
+function impute!(imp::DropObs, table)
+    imp.context() do c
+        @assert istable(table)
+        rows = Tables.rows(table)
+
+        # Unfortunately, we'll need to construct a new table
+        # since Tables.rows is just an iterator
+        table = Iterators.filter(rows) do r
```

Comment:
for interest: https://github.com/JuliaCollections/IterTools.jl/issues/30#issuecomment-452723549